### PR TITLE
fixed minor issues for index_add in docs

### PR DESCRIPTION
**Pull request by ankitaS11**
Hi, I'm looking forward to contributing to PyTorch, so starting with a minor fix in the documentation for `index_add`.

Currently, in the documentation for `index_add_` (please see https://pytorch.org/docs/master/generated/torch.Tensor.index_add_.html#torch.Tensor.index_add_):

1. `tensor` attribute was pointing to `torch.tensor` class, which IMO - is (thought may not be a big deal) unintentional.
2. `dim` attribute is pointing to `torch.Tensor.dim`, which again IMO - is unintentional.

This PR suggests a correction for the first point above, to rename `tensor` attribute to `input` so that it doesn't point to `torch.tensor` class. (I've verified that others ops like `scatter` use `input`, so this should not break the consistency in the documentation). I couldn't find an appropriate fix for the second point above, since renaming `dim` to something else will break the consistency (as almost all others op in PyTorch use `dim` as the attribute name).

I may be wrong here, so please let me know if there is any feedback or an alternate fix for this.

_Note:_ I plan to fix this behavior for `index_copy_` (https://pytorch.org/docs/master/generated/torch.Tensor.index_copy_.html#torch.Tensor.index_copy_) once and if this PR is approved.

To the reviewers, please help me tag the correct person who could help review this PR.

cc: @krshrimali @mruberry @zou3519 

cc @brianjo @mruberry

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
Thanks for the contribution, @ankitaS11 - this looks great to me. Fixing this for `index_copy_` next will probably be a good addition.

Minor note: the implementation uses `source` as the variable name, but I think `input` also makes sense. I will let other reviewers take a call on this, and suggest any revisions if required.

Thanks!

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -1768,14 +1768,14 @@ def add_docstr_all(method, docstr):
 
 add_docstr_all('index_add_',
                r"""
-index_add_(dim, index, tensor, *, alpha=1) -> Tensor
+index_add_(dim, index, input, *, alpha=1) -> Tensor
```
Comment (zou3519) [inline torch/_tensor_docs.py:1771]: The actual name of this argument is `index`, according to https://github.com/pytorch/pytorch/blob/7191dd2613051b1ddfe7f5a9cfd0397b1444a116/aten/src/ATen/native/native_functions.yaml#L5650

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -1768,14 +1768,14 @@ def add_docstr_all(method, docstr):
...
 
-Accumulate the elements of :attr:`alpha` times :attr:`tensor` into the :attr:`self`
+Accumulate the elements of :attr:`alpha` times :attr:`input` into the :attr:`self`
```
Comment (zou3519) [inline torch/_tensor_docs.py:1773]: This should just be
```
``input``
```
The `:attr:` is responsible for funny business you noted where the attribute links to something

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -4657,7 +4657,7 @@ def callable(a, b) -> number
 
 add_docstr_all('index_add',
                r"""
-index_add(dim, index, tensor2) -> Tensor
+index_add(dim, index, input) -> Tensor
```
Comment (zou3519) [inline torch/_tensor_docs.py:4660]: The right name for this argument is also `index` it looks like: https://github.com/pytorch/pytorch/blob/7191dd2613051b1ddfe7f5a9cfd0397b1444a116/aten/src/ATen/native/native_functions.yaml#L5659

**Review by zou3519 — COMMENTED** [review]
Please modify the name of the argument to `source`; that is what the actual kwarg is named:
```
x = torch.randn(10)
x.index_add(0, torch.zeros(10, dtype=torch.long), source=x)
```

File: torch/_tensor_docs.py [hunk 4]
```diff
@@ -1784,11 +1784,11 @@ def add_docstr_all(method, docstr):
 
 Args:
     dim (int): dimension along which to index
-    index (IntTensor or LongTensor): indices of :attr:`tensor` to select from
-    tensor (Tensor): the tensor containing values to add
+    index (IntTensor or LongTensor): indices of :attr:`input` to select from
```
Comment (mruberry) [inline torch/_tensor_docs.py:1787]: While we're here the language "IntTensor" and "LongTensor" is deprecated. Index is just a tensor. The requirement that it have the long (int64) or int (int32) dtype can be noted independently of the Python  type

File: torch/_tensor_docs.py (hunk 2 above)
Comment (krshrimali) [reply torch/_tensor_docs.py:1773]: Interesting, just curious, is there any reason `:attr:` is used for other args like `alpha`?

File: torch/_tensor_docs.py (hunk 4 above)
Comment (ankitaS11) [reply torch/_tensor_docs.py]: Thanks, @mruberry for your suggestion. I think this can be added in the same line after " to select from" (added in the recent commit). Does it sound good to you?

File: torch/_tensor_docs.py (hunk 3 above)
Comment (ankitaS11) [reply torch/_tensor_docs.py]: Thanks, @zou3519. Fixed this. :)

**Comment by ankitaS11** [comment]
Hi @mruberry @zou3519 

Just a gentle ping, if you can take a look into this.

File: torch/_tensor_docs.py [hunk 5]
```diff
@@ -1784,11 +1784,12 @@ def add_docstr_all(method, docstr):
 
 Args:
     dim (int): dimension along which to index
-    index (IntTensor or LongTensor): indices of :attr:`tensor` to select from
-    tensor (Tensor): the tensor containing values to add
+    index (Tensor): indices of ``source`` to select from,
+            should have type either `torch.int64` or `torch.int32`
```
Comment (zou3519) [inline torch/_tensor_docs.py:1788]: nit: "should have type" -> "should have dtype". "dtype" is PyTorch/numpy parlance for data type

File: torch/_tensor_docs.py (hunk 2 above)
Comment (zou3519) [reply torch/_tensor_docs.py]: I don't think there's a good reason for it. I used `:attr:` while writing documentation 3+ years ago for PyTorch because the rest of the docs used `:attr:` but because it does weird things I stopped using it.

**Review by zou3519 — COMMENTED** [review]
One minor nit but after that I think we should be good to go on this!

**Comment by ankitaS11** [comment]
Thanks @zou3519 for the suggestion. I have fixed this.

**Review by zou3519 — APPROVED** [review]
thank you!

File: torch/_tensor_docs.py (hunk 2 above)
Comment (krshrimali) [reply torch/_tensor_docs.py]: Oh, interesting. Thanks for the info, I have been a culprit of using it a couple of times in the recent past :D Hence the question. :)

**Review by mruberry — APPROVED** [review]
